front_end/src/backend/downloads.js

```diff
@@ -0,0 +1,137 @@
+// In these functions the 'fileName' parameter variables refer to filenames of '2mib.txt' '64mib.txt' and '256mib.txt'
+const raw_data = `{
+    "northamerica-northeast1":"Montréal",
+    "northamerica-northeast2":"Toronto",
+    "us-central1":"Iowa",
+    "us-east1":"South Carolina",
+    "us-east4":"Northern Virginia",
+    "us-east5":"Columbus",
+    "us-south1":"Dallas",
+    "us-west1":"Oregon",
+    "us-west2":"Los Angeles",
+    "us-west3":"Salt Lake City",
+    "us-west4":"Las Vegas",
+    "southamerica-east1":"São Paulo",
+    "southamerica-west1":"Santiago",
+    "europe-central2":"Warsaw",
+    "europe-north1":"Finland",
+    "europe-southwest1":"Madrid",
+    "europe-west1":"Belgium",
+    "europe-west2":"London",
+    "europe-west3":"Frankfurt",
+    "europe-west4":"Netherlands",
+    "europe-west6":"Zürich",
+    "europe-west8":"Milan",
+    "europe-west9":"Paris",
+    "asia-east1":"Taiwan",
+    "asia-east2":"Hong Kong",
+    "asia-northeast1":"Tokyo",
+    "asia-northeast2":"Osaka",
+    "asia-northeast3":"Seoul",
+    "asia-south1":"Mumbai",
+    "asia-south2":"Delhi",
+    "asia-southeast1":"Singapore",
+    "asia-southeast2":"Jakarta",
+    "australia-southeast1":"Sydney",
+    "australia-southeast2":"Melbourne"
+}`;
+let regions_json = JSON.parse(raw_data);
+
+
+async function timed_download(fileName, bucket) {
```

Comment:
I believe our standard is to use `camelCase` as opposed to `snake_case` for JS/TS.